### ISSUE TITLE
starting an agent with debug=false that was previously set with debug=true keeps endpoints in debug mode

### ISSUE THREAD

**Issue by aanm** [cilium/cilium#8035]
steps:
1) start agent with `debug: true`
2) start endpoints
3) verify the endpoints are running in debug
4) restart agent with `debug: false`
5) verify the endpoints, after being restored, are still running in debug

version tested: v1.5.1
```
2019-05-15T12:39:07.236948575Z level=info msg="  --debug='false'" subsys=daemon
...
2019-05-15T12:39:09.150642457Z level=info msg="Restoring endpoints..." subsys=daemon
...
2019-05-15T12:39:09.161722236Z level=debug msg="Endpoint restoring" code=OK containerID=ef6beb0112 datapathPolicyRevision=0 desiredPolicyRevision=0 endpointID=1363 endpointState=restoring identity=31572 ipv4=10.0.2.104 ipv6= k8sPodName=default/migrate-svc-client-001-qgj5n policyRevision=0 subsys=endpoint type=0
2019-05-15T12:39:09.161758602Z level=debug msg="Endpoint restoring" code=OK containerID=2364ca2d59 datapathPolicyRevision=0 desiredPolicyRevision=0 endpointID=1545 endpointState=restoring identity=41409 ipv4=10.0.2.10 ipv6= k8sPodName=default/migrate-svc-client-003-ldm6k policyRevision=0 subsys=endpoint type=0
...
2019-05-15T12:39:22.518299501Z level=debug msg="flushing old PolicyMap" containerID=6ff258c4d3 datapathPolicyRevision=0 desiredPolicyRevision=0 endpointID=3237 identity=39272 ipv4=10.0.2.51 ipv6= k8sPodName=default/migrate-svc-server-004-p4dkr subsys=endpoint
2019-05-15T12:39:22.520203096Z level=info msg="Rewrote endpoint BPF program" containerID=3698cb2420 datapathPolicyRevision=0 desiredPolicyRevision=330 endpointID=86 error="<nil>" identity=27895 ipv4=10.0.2.59 ipv6= k8sPodName=kube-system/heapster-v1.6.0-beta.1-84464d8966-9vnc7 subsys=endpoint
2019-05-15T12:39:22.520234579Z level=debug msg="Waiting for proxy updates to complete..." containerID=3698cb2420 datapathPolicyRevision=0 desiredPolicyRevision=330 endpointID=86 identity=27895 ipv4=10.0.2.59 ipv6= k8sPodName=kube-system/heapster-v1.6.0-beta.1-84464d8966-9vnc7 subsys=endpoint
...
```

**Comment by joestringer** [comment]
Ah, interesting. So this option is trusted from the restore of old endpoints rather than regenerating the endpoints upon restore using the current daemon options.

**Comment by aanm** [comment]
Still happening.

**Comment by fristonio** [comment]
I was able to reproduce it for v1.5.1 following the steps mentioned by Andre. But this has been fixed in v1.8/master. [This commit](https://github.com/cilium/cilium/commit/77583f4e49f40f70a4272e36131518d54b5528d4#diff-87ce3efe448980dc74dd709641782ffdR91-R96) fixed the issue.
Thanks, @aanm for helping with the investigation. :rocket:
